### PR TITLE
Add Assistant icon to the Content Wizard toolbar #7334

### LINKED ISSUE #7334: Add AI Operator icon to the Content Wizard toolbar
Page controller of Content Studio's editor page should verify if the Saga app is installed **and** properly configured, in which case it should inject the toggle icon inside the header toolbar which will turn the Assistant UI on and off.

Image

### PR DESCRIPTION
**Pull request by edloidas**

### CODE REVIEW

File: modules/app/src/main/resources/admin/tools/main/main.html [hunk 1]
```diff
@@ -33,6 +33,12 @@
     <link rel="stylesheet" type="text/css" href="{{assetsUri}}/admin/common/styles/lib.css">
     <link rel="stylesheet" type="text/css" href="{{assetsUri}}/styles/contentlib.css">
     <script type="text/javascript" src="{{assetsUri}}/admin/common/js/lib.js"></script>
+
+    {{#hasSaga}}
+    <!-- Append the Saga -->
+    <link rel="stylesheet" type="text/css" href="{{sagaAssetUrl}}/style.css">
+    <script type="text/javascript" src="{{sagaAssetUrl}}/index.js" async></script>
+    {{/hasSaga}}
```
Comment (alansemenov) [inline modules/app/src/main/resources/admin/tools/main/main.html]:
```suggestion

    {{#aiAssistantEnabled}}
    <!-- Append the AI Assitant -->
    <link rel="stylesheet" type="text/css" href="{{aiAssistantAssetUrl}}/style.css">
    <script type="text/javascript" src="{{aiAssistantAssetUrl}}/index.js" async></script>
    {{/aiAssistantEnabled}}
```

File: modules/app/src/main/resources/admin/tools/main/main.js [hunk 2]
```diff
@@ -1,15 +1,18 @@
 const admin = require('/lib/xp/admin');
+const appLib = require('/lib/xp/app');
 const mustache = require('/lib/mustache');
 const portal = require('/lib/xp/portal');
 const i18n = require('/lib/xp/i18n');
 
+const SAGA_APP_KEY = 'com.enonic.app.saga';
```
Comment (alansemenov) [inline modules/app/src/main/resources/admin/tools/main/main.js]:
```suggestion
const AI_ASSISTANT_APP_KEY = 'com.enonic.app.saga';
```

File: modules/app/src/main/resources/admin/tools/main/main.js [hunk 3]
```diff
@@ -1,15 +1,18 @@
...
+
 exports.renderTemplate = function (path, params) {
     const view = resolve('./main.html');
     const toolUri = admin.getToolUrl(app.name, 'main');
-    const isBrowseMode = path === toolUri;
     const enableSecurityPolicy = app.config['contentSecurityPolicy.enabled'] !== 'false';
 
-    params.isBrowseMode = isBrowseMode;
+    params.isBrowseMode = path === toolUri;
+    params.hasSaga = appLib.get({key: SAGA_APP_KEY}) != null;
```
Comment (alansemenov) [inline modules/app/src/main/resources/admin/tools/main/main.js]:
```suggestion
    params. aiAssistantEnabled = appLib.get({key: SAGA_APP_KEY}) != null;
```

File: modules/app/src/main/resources/admin/tools/main/main.js [hunk 4]
```diff
@@ -38,6 +41,7 @@ exports.getParams = function () {
             bundles: ['i18n/phrases'],
             locale: admin.getLocales()
         }),
+        sagaAssetUrl: portal.assetUrl({application: SAGA_APP_KEY}),
```
Comment (alansemenov) [inline modules/app/src/main/resources/admin/tools/main/main.js]:
```suggestion
        aiAssistantAssetUrl: portal.assetUrl({application: SAGA_APP_KEY}),
```

File: modules/lib/src/main/resources/assets/styles/wizard/content-wizard-toolbar.less [hunk 5]
```diff
@@ -181,6 +181,11 @@
     max-width: 120px;
     margin-right: 5px;
   }
+
+  .saga-container {
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/styles/wizard/content-wizard-toolbar.less]:
```suggestion
  .ai-assistant-container {
```

File: modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardToolbar.ts [hunk 6]
```diff
@@ -21,6 +21,10 @@ import {ContentActionCycleButton} from './ContentActionCycleButton';
 import {ContentWizardToolbarPublishControls} from './ContentWizardToolbarPublishControls';
 import {WorkflowStateManager, WorkflowStateStatus} from './WorkflowStateManager';
 
+interface Saga {
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardToolbar.ts]:
```suggestion
interface AIAssistant {
```

File: modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardToolbar.ts [hunk 7]
```diff
@@ -127,6 +132,17 @@ export class ContentWizardToolbar
...
+        const SAGA = window['Saga'] as Saga;
+        if (!SAGA) {
+            return;
+        }
+
+        const sagaContainer = new DivEl('saga-container');
+        this.addElement(sagaContainer);
+        SAGA.renderLaunchButton(sagaContainer.getHTMLElement());
+    }
+
```
Comment (alansemenov) [inline modules/lib/src/main/resources/assets/js/app/wizard/ContentWizardToolbar.ts]:
```suggestion
    private addAIAssistantButton(): void {
        const AI = window['Enonic_AI'] as AIAssistant;
        if (!AI) {
            return;
        }

        const aiAssistantContainer = new DivEl('ai-assistant-container');
        this.addElement(aiAssistantContainer);
        AI.renderLaunchButton(aiAssistantContainer.getHTMLElement());
    }

```

**Review by alansemenov — CHANGES_REQUESTED** [review]
Let's stay name-agnostic in the code. "Saga" will most likely change to something else before we release.

**Review by alansemenov — APPROVED** [review]

**Review by ashklianko — APPROVED** [review]

**Pull request merged**